Frontend/pages/components/CodeSmellList.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+import Card from "react-bootstrap/Card";
+import InteractiveList from "./InteractiveList";
+
+const CodeSmellList = () => {
+  const items = [
+    { id: 1, label: "Option 1" },
+    { id: 2, label: "Option 2" },
+    { id: 3, label: "Option 3" },
+  ];
+
+  const [selectedItemId, setSelectedItemId] = useState<
+    string | number | undefined
+  >();
+
+  const handleItemClick = (itemId: string | number) => {
```

Comment:
Nie lepiej brać typing od id z ListItem?

import ListItem from "./InteractiveList";
...
const handleItemClick = (itemId: ListItem["id"]) = {
    ...
}

Reply:
Niee, wg mnie tak jest też git. Teraz ListItem jest internal typem  nie widocznym na zewnątrz nie dokłada to wiele kodu żeby napisać typ a jest